credentials/src/main/java/me/uport/sdk/credentials/Credentials.kt

```diff
@@ -147,6 +148,42 @@ class Credentials(
     }
 
 
+    /**
+     * Verify and return profile from a
+     * [Selective Disclosure Response JWT](https://github.com/uport-project/specs/blob/develop/messages/shareresp.md).
+     *
+     * @param token **REQUIRED** The JWT response token from a selective disclosure request
+     *
+     * @return a [UportProfile] object
+     */
+    @Suppress("TooGenericExceptionCaught")
+    suspend fun verifyDisclosure(token: String): UportProfile {
+
+        val (_, payload, _) = JWTTools().decode(token)
+
+        val valid = mutableListOf<JwtPayload>()
+        val invalid = mutableListOf<String>()
+
+        payload.verified?.forEach {
+            try {
+                valid.add(JWTTools().verify(it))
+            } catch (e: Exception) {
+                e.printStackTrace()
+                invalid.add(it)
+            }
+        }
+
+        return UportProfile(
+                payload.iss,
+                payload.net,
+                valid,
+                invalid,
+                payload.own?.get("email"),
+                payload.own?.get("name")
+        )
```

Comment:
The `extra` field of the `UportProfile` data class should be filled here as well.

Basically it should be a copy of the payload map extracted using `JWTTools().decodeRaw(token)`